class/class.2.1/hello.cpp

```diff
@@ -0,0 +1,84 @@
+#include <iostream>
+#include <fstream>
+#include <unordered_map>
+#include <string>
+
+const std::string FILENAME = "user.txt";
+
+class UserStats {
+public:
+    void load() {
+        std::ifstream infile(FILENAME);
+        if (!infile) {
+            std::cerr << "Failed to open file for reading.\n";
+            return;
+        }
+
+        std::string name;
+        int count;
+        while (infile >> name >> count) {
+            stats[name] = count;
+        }
+        std::cout << "Loaded stats from file.\n";
+    }
+
+    void save() const {
+        std::ofstream outfile(FILENAME);
+        if (!outfile) {
+            std::cerr << "Failed to open file for writing.\n";
+            return;
+        }
+
+        for (const auto& [name, count] : stats) {
+            outfile << name << " " << count << "\n";
+        }
+        std::cout << "Saved stats to file.\n";
+    }
+
+    void reset(const std::string& name) {
+        if (name == "bread") {
+            stats.clear();
+            std::cout << "All history cleared.\n";
+        } else {
+            stats.erase(name);
+            std::cout << "Statistics reset for " << name << ".\n";
+        }
+    }
+
+    void greet(const std::string& name) {
```

Comment:
here we have a mix of managing data and interaction with user which is a part of business logic. It is better to have single responsibility for the class members. Therefore it is better to have some method like incrementFor which will just manage the data and it can return updated counter as a result. Based on that higher level of code can print a proper message based on business logic.